### PR TITLE
Copy data modules and utility from Stoa

### PR DESCRIPTION
**Pull request by MichaelKim20**
I copied data modules and some utilities from Stoa into SDK.

Relates to https://github.com/bpfkorea/stoa/issues/180

### CODE REVIEW

**Comment by Geod24** [comment]
Yes, this is the way to go. And then we can make Stoa depends on this project and remove the duplication.

**Comment by Geod24** [comment]
The changes here are reverting https://github.com/bpfkorea/boa-sdk-ts/pull/53 which doesn't sound right.

**Comment by MichaelKim20** [comment]
> The changes here are reverting #53 which doesn't sound right.

The previous version was compiled into Node version 15.
And it doesn't match package.json.
I just installed Node version 15 and modified it to package-lock.json matching package.json.

**Comment by MichaelKim20** [comment]
> > The changes here are reverting #53 which doesn't sound right.
> 
> The previous version was compiled into Node version 15.
> And it doesn't match package.json.
> I just installed Node version 15 and modified it to package-lock.json matching package.json.
> I think it is problematic to commit only package-lock.json. From now on, we'd better commit with package.json.

I think it's because of the version difference.
Now we need to use Node version 15.3 or higher.

**Comment by Geod24** [comment]
> The previous version was compiled into Node version 15.
> And it doesn't match package.json.

Yep, that's what I feared, going to revert it then.

> I think it is problematic to commit only package-lock.json. From now on, we'd better commit with package.json.

I'm not sure what you mean.

File: src/modules/data/Height.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import { SmartBuffer } from 'smart-buffer';
+import {Utils} from "../..";
```
Comment (TrustHenry) [inline src/modules/data/Height.ts:15]:
```suggestion
import { Utils } from "../..";
```

File: src/modules/utils/Utils.ts [hunk 2]
```diff
@@ -109,4 +109,42 @@ export class Utils
...
+        let hi = Number(value >> BigInt(32) & BigInt(0xffffffff));
+        buffer[4] = hi;
+        hi = hi >> 8;
+        buffer[5] = hi;
+        hi = hi >> 8;
+        buffer[6] = hi;
+        hi = hi >> 8;
+        buffer[7] = hi;
+    }
+}
```
Comment (Geod24) [inline src/modules/utils/Utils.ts]: Why?
There is https://www.geeksforgeeks.org/node-js-buffer-writebiguint64le-method/
It's even used in Stoa.

File: src/modules/data/KeyPair.ts [hunk 3]
```diff
@@ -151,6 +151,15 @@ export class PublicKey
...
     }
+
+    /**
+     * The public key consisting of zero values for all bytes.
+     * @returns The instance of PublicKey
+     */
+    static get init(): PublicKey
+    {
+        return new PublicKey(Buffer.alloc(SodiumHelper.sodium.crypto_sign_PUBLICKEYBYTES));
+    }
```
Comment (Geod24) [inline src/modules/data/KeyPair.ts:169]: Are those really needed ? It's only used to construct invalid object, which only seems to happen in tests that don't bother creating valid objects.

File: tests/BOAClient.test.ts [hunk 4]
```diff
@@ -168,17 +168,9 @@ describe ('BOA Client', () =>
...
     {
-        boasdk.SodiumHelper.init()
-            .then(() =>
-            {
-                doneIt();
-            })
-            .catch((err: any) =>
-            {
-                doneIt();
-            });
```
Comment (Geod24) [inline tests/BOAClient.test.ts:181]: Thanks! 👍

File: package-lock.json [hunk 5]
```diff
@@ -1,13 +1,7400 @@
 {
   "name": "boa-sdk-ts",
-  "version": "0.0.7",
-  "lockfileVersion": 1,
+  "version": "0.0.8",
+  "lockfileVersion": 2,
```
Comment (Geod24) [inline package-lock.json:4]: So you originally reverted it, and I agree it was good. I went ahead and reverted the PR. Why are you re-adding it now ?

**Review by Geod24 — CHANGES_REQUESTED** [review]
Few fixes needed. Last commit seems like it can be removed.
Another English thing: You fix the document*ation*, not the document itself.

**Comment by MichaelKim20** [comment]
I tried to separate commit from this PR and make another PR, but you were already reviewing it, so I put it back.

File: package-lock.json (hunk 5 above)
Comment (MichaelKim20) [reply package-lock.json:4]: github CI on Linux, when Node is version 15
Modules that are not made of node version 15 cannot be auto-installed with a flag address.
So I had to set the node version to 15.
We created a new package-lock.json by deleting the package-lock.json that already exists and installing npm.
This is the only way to get through Stoa's CI.
And I installed node 15 on Linux and tested it or got the same result.

File: src/modules/utils/Utils.ts (hunk 2 above)
Comment (MichaelKim20) [reply src/modules/utils/Utils.ts]: SDK should also be used in the browser, but some browsers does not support 'writeBigUInt64LE'.

File: src/modules/data/Height.ts (hunk 1 above)
Comment (MichaelKim20) [reply src/modules/data/Height.ts]: I fixed it, Thank you for review

**Review by TrustHenry — APPROVED** [review]
LGTM. @Geod24  except for comments.

File: src/modules/data/KeyPair.ts (hunk 3 above)
Comment (MichaelKim20) [reply src/modules/data/KeyPair.ts:170]: You're right. Let's get rid of it and find another way.

**Comment by MichaelKim20** [comment]
> Few fixes needed. Last commit seems like it can be removed.
> Another English thing: You fix the document_ation_, not the document itself.

The last commit was removed.

**Comment by Geod24** [comment]
Urgh, I didn't realize the `package-lock.json` 10k LoC diff was still here... Can you guys please make sure your configurations are in sync so we stop this yoyo ?

**Pull request merged**